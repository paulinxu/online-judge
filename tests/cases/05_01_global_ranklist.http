###
# Send request:
GET http://127.0.0.1:12345/contests/0/ranklist HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1466
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:18 GMT
# 
# [{"competitive_score_sum":0.0,"highest_scores":[0.0,0.0,0.0],"latest_scores":[0.0,0.0,0.0],"latest_submission":"+262142-12-31T23:59:59.999999999Z","rank":1,"score":0,"scores":[0.0,0.0,0.0],"shortest_times":[[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":0,"user":{"id":0,"name":"root"}}]
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user1"}
###
# Got response:
# HTTP 200
# content-length: 23
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:18 GMT
# 
# {"id":1,"name":"user1"}
###
# Send request:
GET http://127.0.0.1:12345/contests/0/ranklist HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 2932
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:18 GMT
# 
# [{"competitive_score_sum":0.0,"highest_scores":[0.0,0.0,0.0],"latest_scores":[0.0,0.0,0.0],"latest_submission":"+262142-12-31T23:59:59.999999999Z","rank":1,"score":0,"scores":[0.0,0.0,0.0],"shortest_times":[[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":0,"user":{"id":0,"name":"root"}},{"competitive_score_sum":0.0,"highest_scores":[0.0,0.0,0.0],"latest_scores":[0.0,0.0,0.0],"latest_submission":"+262142-12-31T23:59:59.999999999Z","rank":1,"score":0,"scores":[0.0,0.0,0.0],"shortest_times":[[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":0,"user":{"id":1,"name":"user1"}}]
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user2"}
###
# Got response:
# HTTP 200
# content-length: 23
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:18 GMT
# 
# {"id":2,"name":"user2"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":2}
###
# Got response:
# HTTP 200
# content-length: 1077
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:18 GMT
# 
# {"cases":[{"id":0,"info":"","result":"Compilation Success","time":0},{"id":1,"info":"","result":"Accepted","time":0},{"id":2,"info":"","result":"Accepted","time":0},{"id":3,"info":"","result":"Accepted","time":0},{"id":4,"info":"","result":"Accepted","time":0},{"id":5,"info":"","result":"Accepted","time":0},{"id":6,"info":"","result":"Accepted","time":0},{"id":7,"info":"","result":"Accepted","time":0},{"id":8,"info":"","result":"Accepted","time":0},{"id":9,"info":"","result":"Accepted","time":0},{"id":10,"info":"","result":"Accepted","time":0}],"created_time":"2024-07-25T15:33:18.523285985Z","id":0,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":2},"updated_time":"2024-07-25T15:33:18.762434113Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 1062
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:18 GMT
# 
# {"cases":[{"id":0,"info":"","result":"Compilation Success","time":0},{"id":1,"info":"","result":"Accepted","time":0},{"id":2,"info":"","result":"Accepted","time":0},{"id":3,"info":"","result":"Accepted","time":0},{"id":4,"info":"","result":"Accepted","time":0},{"id":5,"info":"","result":"Accepted","time":0},{"id":6,"info":"","result":"Accepted","time":0},{"id":7,"info":"","result":"Accepted","time":0},{"id":8,"info":"","result":"Accepted","time":0},{"id":9,"info":"","result":"Accepted","time":0},{"id":10,"info":"","result":"Accepted","time":0}],"created_time":"2024-07-25T15:33:18.764244453Z","id":1,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0},"updated_time":"2024-07-25T15:33:19.030909236Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 407
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:19 GMT
# 
# {"cases":[{"id":0,"info":"","result":"Compilation Success","time":0},{"id":1,"info":"","result":"Accepted","time":0}],"created_time":"2024-07-25T15:33:19.033351523Z","id":2,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":0},"updated_time":"2024-07-25T15:33:19.237853797Z"}
###
# Send request:
GET http://127.0.0.1:12345/contests/0/ranklist HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 4414
# content-type: application/json
# date: Thu, 25 Jul 2024 15:33:19 GMT
# 
# [{"competitive_score_sum":0.0,"highest_scores":[100.0,0.0,100.0],"latest_scores":[100.0,0.0,100.0],"latest_submission":"2024-07-25T15:33:19.033351523Z","rank":1,"score":200,"scores":[100.0,0.0,100.0],"shortest_times":[[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":2,"user":{"id":0,"name":"root"}},{"competitive_score_sum":0.0,"highest_scores":[0.0,100.0,0.0],"latest_scores":[0.0,100.0,0.0],"latest_submission":"2024-07-25T15:33:18.523285985Z","rank":2,"score":100,"scores":[0.0,100.0,0.0],"shortest_times":[[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":1,"user":{"id":2,"name":"user2"}},{"competitive_score_sum":0.0,"highest_scores":[0.0,0.0,0.0],"latest_scores":[0.0,0.0,0.0],"latest_submission":"+262142-12-31T23:59:59.999999999Z","rank":3,"score":0,"scores":[0.0,0.0,0.0],"shortest_times":[[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807],[9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807,9223372036854775807]],"submission_count":0,"user":{"id":1,"name":"user1"}}]
