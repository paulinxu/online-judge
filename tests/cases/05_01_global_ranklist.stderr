[2024-07-25T15:33:17Z INFO  actix_server::builder] starting 16 workers
[2024-07-25T15:33:17Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
[2024-07-25T15:33:18Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET / HTTP/1.1" 404 0 "-" "-" 0.000114
[2024-07-25T15:33:18Z INFO  oj::contest_module::function_get_contests] 0
[2024-07-25T15:33:18Z INFO  oj::contest_module::function_get_contests] [RankInfo { user: User { id: 0, name: "root" }, rank: 1, scores: [0.0, 0.0, 0.0], highest_scores: [0.0, 0.0, 0.0], latest_scores: [0.0, 0.0, 0.0], competitive_score_sum: 0.0, shortest_times: [[9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: +262142-12-31T23:59:59.999999999Z, score: 0, submission_count: 0 }]
[2024-07-25T15:33:18Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/0/ranklist HTTP/1.1" 200 1466 "-" "-" 0.000325
[2024-07-25T15:33:18Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.000111
[2024-07-25T15:33:18Z INFO  oj::contest_module::function_get_contests] 0
[2024-07-25T15:33:18Z INFO  oj::contest_module::function_get_contests] 0
[2024-07-25T15:33:18Z INFO  oj::contest_module::function_get_contests] [RankInfo { user: User { id: 0, name: "root" }, rank: 1, scores: [0.0, 0.0, 0.0], highest_scores: [0.0, 0.0, 0.0], latest_scores: [0.0, 0.0, 0.0], competitive_score_sum: 0.0, shortest_times: [[9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: +262142-12-31T23:59:59.999999999Z, score: 0, submission_count: 0 }, RankInfo { user: User { id: 1, name: "user1" }, rank: 1, scores: [0.0, 0.0, 0.0], highest_scores: [0.0, 0.0, 0.0], latest_scores: [0.0, 0.0, 0.0], competitive_score_sum: 0.0, shortest_times: [[9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: +262142-12-31T23:59:59.999999999Z, score: 0, submission_count: 0 }]
[2024-07-25T15:33:18Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/0/ranklist HTTP/1.1" 200 2932 "-" "-" 0.000475
[2024-07-25T15:33:18Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.000088
[2024-07-25T15:33:18Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 1077 "-" "-" 0.239360
[2024-07-25T15:33:19Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 1062 "-" "-" 0.266911
[2024-07-25T15:33:19Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 407 "-" "-" 0.204803
[2024-07-25T15:33:19Z INFO  oj::contest_module::function_get_contests] 200
[2024-07-25T15:33:19Z INFO  oj::contest_module::function_get_contests] 0
[2024-07-25T15:33:19Z INFO  oj::contest_module::function_get_contests] 100
[2024-07-25T15:33:19Z INFO  oj::contest_module::function_get_contests] [RankInfo { user: User { id: 0, name: "root" }, rank: 1, scores: [100.0, 0.0, 100.0], highest_scores: [100.0, 0.0, 100.0], latest_scores: [100.0, 0.0, 100.0], competitive_score_sum: 0.0, shortest_times: [[9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: 2024-07-25T15:33:19.033351523Z, score: 200, submission_count: 2 }, RankInfo { user: User { id: 2, name: "user2" }, rank: 2, scores: [0.0, 100.0, 0.0], highest_scores: [0.0, 100.0, 0.0], latest_scores: [0.0, 100.0, 0.0], competitive_score_sum: 0.0, shortest_times: [[9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: 2024-07-25T15:33:18.523285985Z, score: 100, submission_count: 1 }, RankInfo { user: User { id: 1, name: "user1" }, rank: 3, scores: [0.0, 0.0, 0.0], highest_scores: [0.0, 0.0, 0.0], latest_scores: [0.0, 0.0, 0.0], competitive_score_sum: 0.0, shortest_times: [[9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807], [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807]], latest_submission: +262142-12-31T23:59:59.999999999Z, score: 0, submission_count: 0 }]
[2024-07-25T15:33:19Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /contests/0/ranklist HTTP/1.1" 200 4414 "-" "-" 0.000470
